samples/parallel-coords-polar

KacperMadej opened the pull request:
Added parallel coordinates polar demo.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1207097387397302

hubertkozik commented on samples/highcharts/demo/parallel-coordinates-polar/demo.html:
```diff
@@ -0,0 +1,14 @@
+<script src="https://code.highcharts.com/highcharts.js"></script>
+<script src="https://code.highcharts.com/highcharts-more.js"></script>
+<script src="https://code.highcharts.com/modules/parallel-coordinates.js"></script>
+<script src="https://code.highcharts.com/modules/exporting.js"></script>
+<script src="https://code.highcharts.com/modules/accessibility.js"></script>
```
There are styles declared for the data table, but currently, the `export-data` module is not loaded. I assume it should be loaded 😁

hubertkozik commented on samples/highcharts/demo/parallel-coordinates-polar/demo.js:
```diff
@@ -0,0 +1,164 @@
+        min: -0.5,
+        max: 2.5
+    }],
+    series: data.map(function (set, i) {
```
As GH Actions says: `i` variable is defined but never used.

hubertkozik commented on samples/highcharts/demo/parallel-coordinates-polar/readme.md:
```diff
@@ -0,0 +1,7 @@
+[Parallel coordinates](https://www.highcharts.com/docs/chart-and-series-types/parallel-coordinates-chart) are often used to visualize multidimensional numerical datasets. In this demo, five key properties of selected elements from the periodic table are displayed on one chart. Each property has its own radial axis, showcasing trends and variations across different elemental groups and states.
+
+####  Tip
+Use the following module ***/modules/parallelcoordinates.js*** to create the parallel coordinates.
```
It should be ***/modules/parallel-coordinates.js*** with a dash.

nasin commented:
Hi, I made some changes in the CSS and the JS files. I changed the stroke color on hover so that it matches the series color and made the axis lines dashed. Then, in the JS file, I changed the size and color of the axis labels.

nasin reviewed (approved):
I made some style changes in the CSS and JS files

KacperMadej commented:
I moved all possible to move CSS options into the chart config to avoid problems with exporting - CSS are used for exporting only in `styledMode`. Usage of colors outside of `Highcharts.getOptions().colors` for series will break the demos custom colors palette, so I have removed this for now. The result is as close as possible to the suggested design without using too much of extra code IMO.

TorsteinHonsi commented on samples/highcharts/demo/parallel-coordinates-polar/demo.js:
```diff
@@ -0,0 +1,195 @@
+const data = [
+    // Name (Sybol), Atomic Number, Atomic Radius, Boiling Point, Density, Phase
```
```suggestion
    // Name (Symbol), Atomic Number, Atomic Radius, Boiling Point, Density, Phase
```

TorsteinHonsi commented on samples/highcharts/demo/parallel-coordinates-polar/demo.js:
```diff
@@ -0,0 +1,195 @@
+            }
+        },
+        gridLineWidth: 0,
+        plotLines: Array.from({ length: 5 }, (_, value) => ({
```
```suggestion
        // Display custom grid lines in the center of the category
        plotLines: Array.from({ length: 5 }, (_, value) => ({
```

TorsteinHonsi commented on samples/highcharts/demo/parallel-coordinates-polar/demo.js:
```diff
@@ -0,0 +1,195 @@
+    }, {
+        // Density
+        labels: {
+            format: '{value:.2f} g/cm³',
```
Arguably, two decimals are not needed here because the precision is handled internally.
```suggestion
            format: '{value} g/cm³',
```

TorsteinHonsi commented on samples/highcharts/demo/parallel-coordinates-polar/demo.js:
```diff
@@ -0,0 +1,195 @@
+        min: 0
+    }, {
+        // Density
+        labels: {
```
On my screen, these axis labels don't render. If I increase the `chart.height` to 500, they do. I tried adjusting `tickPixelInterval` to make them render at default height, but that doesn't seem to do anything. What's going on? Are we overriding `tickPixelInterval` or something for parallel coords (or polar)?

KacperMadej replied:
In `chart.parallelAxes` parallel coords. Showing of edge (1st & last) labels is disabled and for long labels it might be the case that nothing is shown because there are only 2 labels. There seems to be a bug with resize and axis labels ([demo](https://jsfiddle.net/BlackLabel/msdr1ohx/26/)). I will look into this.

TorsteinHonsi replied:
Okay, that makes sense. Feels like a bug, but I don't really have a good idea for a better default behaviour.

KacperMadej replied:
Internally calculated tickInterval from unsquish seems to be unstable but it doesn't affect the demo any more as far as I can see (after the recent changes), so we could handle those tasks separately IMO.

KacperMadej replied:
Y axis labels have size `11px` and `fontMetrics` running on init (+ 2nd run due to responsive rules) doesn't find the tick label (there are no ticks yet) and falls back to chart's default `16px` causing instability when running on reflow/resize after the tick labels are found and placed correctly.

Minimal bug report demo: https://jsfiddle.net/BlackLabel/1vq9g4wd/
The bug reported here - https://github.com/highcharts/highcharts/issues/22027

KacperMadej replied:
@TorsteinHonsi If the font for the axis labels will be default then this won't be visible in this demo, but I can no longer recreate the problem. With the current options it shows fine for me but this might depend on OS font rendering as the changes are between some floats in range 0-2 and rounding impacts the outcome.

TorsteinHonsi reviewed (changes requested):
Thanks!
